content/doc/tutorials/build-a-dotnet-web-app-with-jenkins.adoc

```diff
@@ -0,0 +1,363 @@
+
+[.boxshadow]
+image:tutorials/dotnet/test-coverage-result.png[alt="Test coverage result",width=100%]
+
+=== Add a final deliver stage to your Pipeline
+
+. Go back to your text editor/IDE and ensure your `Jenkinsfile` is open.
+. Copy and paste the following Declarative Pipeline syntax immediately under the `Test` stage of your `Jenkinsfile`:
++
+[source,groovy]
+----
+        stage('Deliver') {
+            steps {
+                sh 'dotnet publish SimpleWebApi --no-restore -o published'
+            }
+            post {
+                success {
+                    archiveArtifacts 'published/*.*'
+                }
+            }
+        }
+----
+Add a `skipStagesAfterUnstable` option, resulting in:
++
+[source,groovy]
+----
+pipeline {
+    agent any
+    options {
+        skipStagesAfterUnstable()
+    }
+    stages {
+        stage('Build') {
+            steps {
+                sh 'dotnet restore'
+                sh 'dotnet build --no-restore'
+            }
+        }
+        stage('Test') {
+            steps {
+                sh 'dotnet test --no-build --no-restore --collect "XPlat Code Coverage"'
+            }
+            post {
+                always {
+                    recordCoverage(tools: [[parser: 'COBERTURA', pattern: '**/*.xml']], sourceDirectories: [[path: 'SimpleWebApi.Test/TestResults']])
+                }
+            }
+        }
+        stage('Deliver') { // <1>
+            steps {
+                sh 'dotnet publish SimpleWebApi --no-restore -o published'  // <2>
+            }
+            post {
+                success {
+                    archiveArtifacts 'published/*.*' // <3>
+                }
+            }
+        }
+    }
+}
+----
+<1> Defines a new stage called `Deliver` that appears on the Jenkins UI.
+<2> This link:/doc/pipeline/steps/workflow-durable-task-step/#code-sh-code-shell-script[`sh`] step executes the `dotnet publish` command to publish your .NET web app into executable and store it into the `published` directory.
+<3> This link:/doc/pipeline/steps/workflow-durable-task-step/#code-archiveartifacts-code-archive-the-artifacts[`archiveArtifacts`] step archives the `published` directory as a build artifact.
+
+. Save your updated `Jenkinsfile` and commit it to your local `simple-dotnet-web-app` Git repository.
++
+.. `git stage .`
+.. `git commit -m "Add 'Deliver' stage"`
+.. `git push` to push your changes to your forked repository on GitHub, so it can be picked up by Jenkins.
+
+. Return to the *Dashboard* and select *dotnet-tutorial*.
+. Select *Build Now* on the left. After a while, a new column titled *Deliver* will appear in the Stage View.
++
+[.boxshadow]
+image:tutorials/dotnet/deliver-stage-added.png[alt="Deliver stage runs successfully",width=100%]
+
+You can select *#3* (or the number representing your last build) in the *Builds* widget to see the details of the build.
```

Comment:
```suggestion
You can select *#3* (or the number representing your last build) in the *Stage View* or *Builds* widget to see the details of the build.
```
Same as above, being explicitly clear where to find the links. Would it be possible to include the build widget in this screenshot too?